test-engine-core-modules/src/rmodel/module_tests/plugin_test.py

```diff
@@ -0,0 +1,174 @@
+    """
+
+    def __init__(self, model_path: str, discover_path: Path):
+        # Other variables
+        self._base_path: Path = discover_path
+
+        # Store the input arguments as private vars
+        self._model_path: str = str(self._base_path / model_path)
+
+        # Default for instances
+        self._model_instance: Union[None, IModel] = None
+        self._model_serializer_instance: Union[None, ISerializer] = None
+
+        self._expected_model_algorithm = "rpy2.robjects.vectors.ListVector"
+        self._expected_model_plugin_type = ModelPluginType.R
+
+    @time_class_method
+    def run(self) -> None:
+        """
+        A function to run the plugin test with the provided arguments.
+        """
+        try:
+            error_count = 0
+            error_message = ""
+
+            # Load all the core plugins and the model plugin
+            PluginManager.discover(str(self._base_path))
+
+            # Get the model instance
+            (
+                self._model_instance,
+                self._model_serializer_instance,
+                error_message,
+            ) = PluginManager.get_instance(
+                PluginType.MODEL, **{"filename": self._model_path}
+            )
+
+            # Perform model instance setup
+            is_success, error_messages = self._model_instance.setup()
+            if not is_success:
+                raise RuntimeError(
+                    f"Failed to perform model instance setup: {error_messages}"
+                )
+
+            # Run different tests on the model instance
+            test_methods = [
+                self._validate_metadata,
+                self._validate_plugin_type,
+                self._validate_model_supported,
+                self._
+            ]
+
+            for method in test_methods:
+                tmp_count, tmp_error_msg = method()
+                error_count += tmp_count
+                error_message += tmp_error_msg
+
+            # Perform cleanup
+            self._model_instance.cleanup()
+
+            if error_count > 0:
+                print(f"Errors found while running tests. {error_message}")
+                sys.exit(-1)
+            else:
+                print("No errors found. Test completed successfully.")
+                sys.exit(0)
+
+        except Exception as error:
+            # Print and exit with error
+            print(f"Exception found while running tests. {str(error)}")
+            sys.exit(-1)
+
+    def _validate_metadata(self) -> Tuple[int, str]:
+        """
+        A helper method to validate metadata
+
+        Returns:
+            Tuple[int, str]: Returns error count and error messages
+        """
+        error_count = 0
+        error_message = ""
+
+        metadata = self._model_instance.get_metadata()
+        if (
+            metadata.name == "rmodel"
+            and metadata.description
+            == "rmodel supports detecting R models"
```

Comment:
Hi, the metadata here doesn't seem to match the actual metadata. `PluginMetadata(name='rmodel', description='rmodel supports detecting r models', version='0.9.0')`